python_modules/dagster/dagster/_core/definitions/definitions_loader.py

```diff
@@ -45,6 +45,8 @@ class DefinitionsLoadContext:
     User construction of this object is not supported.
     """
 
+    _instance: ClassVar[Optional["DefinitionsLoadContext"]] = None
```

Comment:
I think we can use https://docs.python.org/3/library/contextvars.html here? @alangenfeld has used this before.